components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java

```diff
@@ -215,6 +250,20 @@ private void addThrottledErrorPropertiesToChannel(ChannelHandlerContext ctx) {
         WebSocketUtils.setApiPropertyToChannel(ctx, SynapseConstants.ERROR_MESSAGE, "Message Throttled Out");
     }
 
+    private void addAuthErrorPropertiesToChannel(ChannelHandlerContext ctx) {
+        WebSocketUtils.setApiPropertyToChannel(ctx, SynapseConstants.ERROR_CODE,
+                WebSocketApiConstants.FrameErrorConstants.API_AUTH_GENERAL_ERROR);
+        WebSocketUtils.setApiPropertyToChannel(ctx, SynapseConstants.ERROR_MESSAGE,
+                "Authentication Failure");
+    }
+
+    private void addOtherErrorPropertiesToChannel(ChannelHandlerContext ctx) {
+        WebSocketUtils.setApiPropertyToChannel(ctx, SynapseConstants.ERROR_CODE,
+                WebSocketApiConstants.FrameErrorConstants.OTHER_ERROR);
+        WebSocketUtils.setApiPropertyToChannel(ctx, SynapseConstants.ERROR_MESSAGE,
+                "Unclassified Error");
```

Comment:
same here. why cant we publish the error code and message in the InboundProcessorResponseDTO itself?